modules/http-api/src/main/java/org/mule/service/http/api/domain/message/request/HttpRequestBuilder.java

```diff
@@ -49,6 +52,15 @@ public HttpRequestBuilder setMethod(String method) {
   }
 
   /**
+   * @param method the HTTP method of the {@link HttpRequest} desired. Non null.
+   * @return this builder
+   */
+  public HttpRequestBuilder setMethod(Method method) {
```

Comment:
I'm not sure this is a good idea. The RFC allows extension methods, meaning a server could define it's own specific ones. I'd keep the String signature as well.